.github/workflows/build.yml

```diff
@@ -72,7 +72,7 @@ jobs:
           mkdir dist out
           make build
           zip -r -j out/${{ env.PKG_NAME }}_${{ needs.get-product-version.outputs.product-version }}_linux_${{ matrix.arch }}.zip dist/
-      - uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+      - uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.3
```

Comment:
@jessebot v4 of the upload/download actions are not backwards compatible with the older v3 and so on versions. The `hashicorp/actions-docker-build@v1` integrates v3 of both and they cannot be mixed. It was one of the reasons I was looking at alternatives to get away from the Hashicorp dependency. https://github.com/openbao/openbao/pull/125